### PR TITLE
Sanity check the OpenSearch bundle.

### PR DESCRIPTION
**Pull request by dblock**
Signed-off-by: dblock <dblock@dblock.org>

### Description

Today it's possible to make a change to an OpenSearch bundle manifest and break everything. The simplest mistake is a typo in the GitHub URL. This PR adds a CI workflow and a GitHub action that runs it to ensure that the components are sane.  Future versions can get more crafty in terms of what's being checked (e.g. plugin versions). 

Originally I thought I'd add a dry run or a check option to the build workflow, but I think we're going to get more and more advanced with what we want to do in checks, so adding a new workflow for that.
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/manifests.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      - name: Set up Python ${{ env.PYTHON_VERSION }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+      - name: Install Pipenv and Dependencies
+        run: |
+          python -m pip install --upgrade pipenv wheel
+      - name: OpenSearch 1.1
+        run: |
+          ./bundle-workflow/ci.sh manifests/opensearch-1.1.0.yml
```
Comment (peternied) [inline .github/workflows/manifests.yml:24]: Rather than only run on the "current" manifest, could we have this enforced on all manifest files?

**Review by peternied — APPROVED** [review]
I've got a slight anxiety about this check since it is an integration tests between repositories and thus could break out of band of changes here... however this is tempered by the fact no one should delete branches they want to ship.

File: .github/workflows/manifests.yml (hunk 1 above)
Comment (dblock) [reply .github/workflows/manifests.yml]: Good idea.

Comment (dblock) [reply .github/workflows/manifests.yml]: I figured out how to dynamically list manifest files and parallel run checks on them :)

Screen Shot 2021-09-03 at 6 28 28 PM

**Comment by dblock** [comment]
> I've got a slight anxiety about this check since it is an integration tests between repositories and thus could break out of band of changes here... however this is tempered by the fact no one should delete branches they want to ship.

And this should allow us to quickly locate any such problems!

File: .github/workflows/manifests.yml (hunk 1 above)
Comment (peternied) [reply .github/workflows/manifests.yml]: Amazing! 🚢

**Review by peternied — DISMISSED** [review]

**Review by peternied — APPROVED** [review]

**Review by mch2 — APPROVED** [review]

**Review by saratvemulapalli — APPROVED** [review]
This is awesome :)!

**Pull request merged**